lib/AST/ASTDumper.cpp

```diff
@@ -153,7 +153,11 @@ void RequirementRepr::printImpl(raw_ostream &out, bool AsWritten) const {
 }
 
 void RequirementRepr::print(raw_ostream &out) const {
-  printImpl(out, /*AsWritten=*/false);
+  StreamPrinter printer(out);
+  printImpl(printer, /*AsWritten=*/false);
+}
+void RequirementRepr::print(ASTPrinter &out) const {
+  printImpl(out, /*AsWritten=*/true);
```

Comment:
ASTPrinter seemed like a higher-level construct, but now that I think about it more, is printing as-written even the "higher-level" choice/the right choice for how I'm using it?